jobclient/python/cookclient/__init__.py

```diff
@@ -279,6 +341,47 @@ def close(self):
         if self.__session is not None:
             self.__session.close()
 
+    @staticmethod
+    def _apply_jobspec_defaults(jobspec: dict):
+        """Apply default values to a jobspec.
+
+        This function will add default values to a jobspec if no value is
+        provided. The provided jobspec will be modified in-place.
```

Comment:
Modifying in place seems bad to me. E.g, calling this twice would be a problem. Don't like the refactor.

Reply:
@hdrodz97

Reply:
Fixed them, they no longer modify in-place